### ISSUE TITLE
importing a file and including it as a cli arguments leads to bad error message

### ISSUE THREAD

**Issue by expipiplus1** [shader-slang/slang#5660]
`nullType.slang`:
```slang
import nullTypeModule;

struct S { }

extension S : I { }
```

`nullTypeModule.slang`:
```slang
interface I { } 
```

`slangc -stage compute tests/bugs/nullTypeModule.slang tests/bugs/nullType.slang`

```
tests/bugs/nullType.slang(5): error 39999: ambiguous reference to 'I'
extension S : I { }
              ^
tests/bugs/nullTypeModule.slang(1): note 39999: candidate: interface I
interface I { }
          ^
tests/bugs/nullTypeModule.slang(1): note 39999: candidate: interface I
interface I { }
          ^
```

This "ambiguous" reference refers to the same name

**Comment by bmillsNV** [comment]
@aleino-nv can you help to verify if this issue still occurs? We think you might have fixed something similiar.

**Comment by aleino-nv** [comment]
> [@aleino-nv](https://github.com/aleino-nv) can you help to verify if this issue still occurs? We think you might have fixed something similiar.

Yes, this seems similar to https://github.com/shader-slang/slang/issues/6221. I'll take a look.

**Comment by aleino-nv** [comment]
It still happens today on master (133d705dcaa69f10f7a713605d5080405e49875c), even after the #6221 fix:

```
C:\Users\aleino\Downloads>C:\src\slang\build\Debug\bin\slangc.exe -stage compute nullTypeModule.slang nullType.slang   || exit 1 
nullType.slang(5): error 39999: ambiguous reference to 'I'
extension S : I { }
              ^
nullTypeModule.slang(1): note 39999: candidate: interface I
interface I { }
          ^
nullTypeModule.slang(1): note 39999: candidate: interface I
interface I { }
          ^
(0): error 30015: undefined identifier 'main'.
nullTypeModule.slang(1): error 38000: no function found matching entry point name 'main'
interface I { }
^~~~~~~~~
```

**Comment by aleino-nv** [comment]
Assigning to @bmillsNV, since I probably won't have time to investigate this sprint.

**Comment by kaizhangNV** [comment]
@expipiplus1 
I don't quite understand why we should compile in this way?

```
slangc -stage compute nullTypeModule.slang nullType.slang
```

we should only compile `nullType.slang`, because `nullTypeModule.slang` is imported to `nullType.slang` in the source code.

so I don't know if this is actually legal?
```
slangc -stage compute nullTypeModule.slang nullType.slang
```

because in this way, I think we probably compile the module `nullTypeModule.slang` twice, and of course there will be such duplicated definition.

**Comment by jarcherNV** [comment]
Yes that's my understanding of what's happening here. Perhaps instead we could give a compile error for a module trying to import itself?

**Comment by jarcherNV** [comment]
I have created a PR to fix the module import issue. Prior to my change, it was possible to declare a module and then immediately import it. I don't know if that's intended behavior or not, if it is then we should not take my change.

After my change, the new error reported for the above shaders is:

nullType.slang(3): error 38200: module `nullTypeModule` recursively imports itself
import nullTypeModule;
       ^~~~~~~~~~~~~~

**Comment by jarcherNV** [comment]
I'd be ok with changing this to a warning instead. In this case the duplicated 39999 errors would remain but it would hopefully be obvious enough to the user what the problem is.

**Comment by jarcherNV** [comment]
I have updated this locally to produce a warning instead, but I now realise my test was incorrect. I'll need to write one that compiles two Slang shaders as described above but I'm not sure how to do that, @kaizhangNV do you know if that's possible or is each test a single shader only?

**Comment by jkwak-work** [comment]
(With Cursor), I found that the behavior of the multiple input files to slangc is described in here,
https://shader-slang.org/slang/user-guide/compiling.html#command-line-compilation-with-slangc

It says
> If there are any .slang files, then all of them will be grouped into a single translation unit

So I guess either `import` shouldn't be used, or only one input file should be given to slangc.

**Comment by jarcherNV** [comment]
Yes if you give just the one input file, or don't use import, then this compiles fine. I think we'd still want the warning in case a user does either of those things though right?

**Comment by jarcherNV** [comment]
I have updated the PR to instead report a new warning for a module reporting itself, along with a test matching the above example shaders which checks for the new warning.

**Comment by jarcherNV** [comment]
Moving this back to TODO as I've not had a chance to look at this for a while.

**Comment by gtong-nv** [comment]
This issue might require a breaking change based on the converstaion above and comments in https://github.com/shader-slang/slang/pull/6713

The Original Problem: Running `slangc A.slang B.slang` where `A.slang` imports `B.slang` produces a confusing "ambiguous reference" error that points to the same declaration twice.
The Initial Fix Attempt: PR #6713 tried to fix this by reporting an error/warning when a module imports itself.
@csyonghe  clarified that the real solution is a semantic change in how **slangc** handles multiple input files:

- Current Behavior	
  - All files become one module
  - Import causes "duplicate" error	
- Proposed Behavior
  - Each file becomes its own separate module
  - Modules are deduplicated in composite component type

I think the best next step is to create a new issue to track the breaking change and documentation update.

**Comment by gtong-nv** [comment]
Based on the discussion, we decided to close this issue. This issue should be resolved if we move to use the new compilation API for slangc, instead of 
`input .slang files will be grouped into a single logical translation unit` and using `SlangCompileRequest`.

The effort is tracked in
#4759 
#9183
